crates/test/mock-consensus/src/send_block.rs

```diff
@@ -0,0 +1,55 @@
+//! [`TestNode`] interfaces for sending [`Block`]s.
+
+use {
+    crate::TestNode,
+    tendermint::{
+        v0_37::abci::{ConsensusRequest, ConsensusResponse},
+        Block,
+    },
+    tower::{BoxError, Service},
+    tracing::{info, instrument},
+};
+
+impl<C> TestNode<C>
+where
+    C: Service<ConsensusRequest, Response = ConsensusResponse, Error = BoxError>
+        + Send
+        + Clone
+        + 'static,
+    C::Future: Send + 'static,
+    C::Error: Sized,
+{
+    /// Sends the provided [`Block`] to the consensus service.
+    ///
+    /// Use [`TestNode::block()`] to build a new block.
+    #[instrument(
+        level = "info",
+        skip_all,
+        fields(
+            height = %block.header.height,
+            time = %block.header.time,
+            app_hash = %block.header.app_hash,
+        )
+    )]
+    pub async fn send_block(&mut self, block: Block) -> Result<(), anyhow::Error> {
+        let Block {
+            header,
+            data,
+            evidence: _,
+            last_commit: _,
+            ..
+        } = block;
+
+        info!("sending block");
+        self.begin_block(header).await?;
+        for tx in data {
+            let tx = tx.into();
+            self.deliver_tx(tx).await?;
+        }
+        self.end_block().await?;
+        self.commit().await?;
+        info!("finished sending block");
```

Comment:
:rocket:

(_also, a note for reviewers: this is the right flow of requests, correct?_)

Reply:
Yep, this looks right.

The only thing missing is that there's no `PrepareProposal`/`ProcessProposal`. But since we're directly controlling the contents of the block anyways, we don't really _need_ to execute those, at least at the moment.

Reply:
filed #3826 to track that work in the future. i'll mark this as resolved now! :white_check_mark: